src/Controller/ResourceController.php

```diff
@@ -345,27 +357,41 @@ private function getDirectory(Request $request, $path, $responseFormat, array $v
             ];
         }
 
-        $content = $this->getGraphForPath($request, $path)->serialise($responseFormat, $options);
+        $content = $this->getGraphForPath($request, $requested_path)->serialise($responseFormat, $options);
 
         $headers = [
-            "Last-Modified" => $modifiedTime->format(\DateTime::W3C),
-            "Link" => ["<{$this->LDP_NS}Resource>; rel=\"type\"",
-                       "<{$this->LDP_NS}BasicContainer>; rel=\"type\""],
-            "Vary" => "Accept",
+            "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                       "<".self::LDP_NS."BasicContainer>; rel=\"type\""],
             "Content-Type" => $responseMimeType,
             "Content-Length" => strlen($content)
         ];
 
-        if (!$doGet) {
-            $content = '';
+        if ($request->getMethod() == 'GET') {
+            $response->setContent($content);
```

Comment:
Ups. missing one refactor commit. Should be fixed via https://github.com/trellis-ldp/static-ldp/pull/22/commits/172efea44812f9ea3cb8776cf9674662d63e9eba.
The only way of keeping processing optimal was to really just make use of a common response object and since request was already present in the arguments, the doGet boolean was really extra code.